### ISSUE TITLE
Gene panel Oncoprint issue

### ISSUE THREAD

**Issue by priti88** [cBioPortal/cbioportal#2556]
Sample's coverage is not reflecting correct panel. Screen shot from Genie cbioportal instance:
<img width="1269" alt="screen shot 2017-06-07 at 5 37 15 pm" src="https://user-images.githubusercontent.com/1380237/26931344-2da96362-4c2e-11e7-8428-65d4e745fe91.png">

I am having the same problem in the internal instance and if I look into mysql, the sample_profile table is not getting populated correctly by the importer.

**Comment by ritikakundra** [comment]
I looked at the genie files, all look fine. Looks like samples that don't have mutations in a gene shows up as WXS, or else will show up with the correct panel.

**Comment by jjgao** [comment]
That shouldn't be the case. Looping in @adamabeshouse.

**Comment by tmazor** [comment]
To clarify, it's not that all samples without a mutation in a gene show up with WES for their panel, it's that a subset of samples show up with WES as their panel. As far as I can tell, it's a random subset of cases that show up with WES, whether or not they have a mutation

**Comment by jjgao** [comment]
Maybe they don't have a gene panel assigned? @ritikakundra could you check the data file?

**Comment by tmazor** [comment]
We had this problem with profile data in our internal version and Priti confirmed that they had the correct gene panel assigned.

**Comment by tmazor** [comment]
Priti also just looked up the MSK GENIE case from the screenshot and it is assigned to MSK-IMPACT341

**Comment by tmazor** [comment]
I amend my previous statement. I can't find a case with a mutation that also says WES. But a subset of cases with copy number changes (screenshot below) or no mutation do show up as WES.

![image](https://user-images.githubusercontent.com/10508276/28691294-67b6ccba-72ea-11e7-88b5-d851a15ed39a.png)

**Comment by ritikakundra** [comment]
I just queried 1799 samples with no mutations for some common driver genes, all showed up as WXS. Then I queried the 17005 samples with some mutation data for the same genes and they all showed the right panel. I do believe it's for all samples that were sequenced but have no mutation data.

**Comment by tmazor** [comment]
Hmmm ok. I don't know what's going on then. When I load up the GENIE data, most of the samples without a mutation have a gene panel and only a small number show up as WES

http://www.cbioportal.org/genie/index.do?session_id=59760665498e5df2e292f0c2&show_samples=false&
![image](https://user-images.githubusercontent.com/10508276/28692492-c9e2b058-72ee-11e7-887e-097e02353a56.png)

**Comment by ritikakundra** [comment]
@tmazor Right, cos this sample (102) has other mutations. The 1799 samples am talking about have no mutations in any genes.

**Comment by tmazor** [comment]
Oh! Now I understand! Thanks @ritikakundra!

**Comment by jjgao** [comment]
@ritikakundra @tmazor thanks for figuring it out. Handing it over to @adamabeshouse :)

**Comment by adamabeshouse** [comment]
so the way that this works is the frontend sends a query to the backend, asking about the queried genes and the queried samples. The backend gives back a list of gene panels, with sequenced samples and genes of the queried that are sequenced. So for every (sample * gene), there are three possibilities:
(1) There is a gene panel for that sample, which includes that gene -> we infer "sequenced", give link to gene panel
(2) There is a gene panel for that sample, but it does not include that gene -> we infer "not sequenced" (light gray)
(3) There is no gene panel in the list that includes that sample -> we infer WXS

I'm a bit at a loss about how to handle this - it seems to me the problem is that the data does not reflect this system. Is there a way to modify the data in production to fit into this system?

**Comment by adamabeshouse** [comment]
Update:

So I looked at the query for gene panel information, and there's nothing wrong with the query or the return value, except that the return value is missing the samples, which is why they're showing up as WXS even though they have specific gene panel associated to them. The query is done by gene and genetic profile, not by gene and samples as I thought before, so it seems to me that the fact that this issue is only happening for samples without any mutations in any gene indicates that something is not handled right in the gene panel service, that only samples with mutations in the given profile are considered, rather than all the samples sequenced for that profile.

Looping in @sheridancbio who worked on this service.
In short, @sheridancbio: can you think of any reason why, when we pass to the service the genetic profile to query, that only samples that have mutations would be considered and returned, rather than all samples sequenced in that profile?

**Comment by sheridancbio** [comment]
Ok, I'll review and take a look. I'm starting to remember that we carefully went over the topic of using the "no gene panel" return value as being synonymous with whole genome/exome sequencing. Maybe in the PR's for the service/persistence layer there is discussion which will help clarify what is going on.

When you (Adam) say that the return value is "missing the samples" do you mean that only the samples which have no mutation in any gene are missing? If that is the case then I would guess the problem is in the SQL joining logic (one of the inner joins needs to be changed to an outer join)

**Comment by adamabeshouse** [comment]
@sheridancbio that's right, the gene panel that the sample is associated to gets returned from the service, but samples that dont have any mutations (in any gene) are missing from the sample list for that panel.

**Comment by tmazor** [comment]
@adamabeshouse @sheridancbio I lost track of this conversation - did this issue get resolved?

**Comment by tmazor** [comment]
@jjgao any updates on this?

**Comment by jjgao** [comment]
@adamabeshouse @sheridancbio do we need to make any change to the gene panel API to fix this problem? cc'ing @ersinciftci

**Comment by ersinciftci** [comment]
@jjgao we are using the new Gene Panel API now, so the old one is not relevant anymore. I feel like things should work properly now with this change https://github.com/cBioPortal/cbioportal/issues/3628 (or maybe even before this change, but this change made things easier), if I understand the problem correctly.

**Comment by jjgao** [comment]
I still see the same issue in the genie portal. @adamabeshouse maybe could you look into this?

**Comment by adamabeshouse** [comment]
unless the data/ontology issue @sheridancbio was looking into has been resolved, we're still going to see this issue

**Comment by adamabeshouse** [comment]
I think it will be fixed by https://github.com/cBioPortal/cbioportal-frontend/pull/970 but for various reasons its not possible to test until next time we deploy to genie

**Comment by tmazor** [comment]
Looks like its working now - thanks all!

**Comment by adamabeshouse** [comment]
whoo!!!! 🎉  got to love when an issue from june '17 closes

**Comment by jjgao** [comment]
it's captured in another issue: https://github.com/cBioPortal/cbioportal/issues/3965
